### PR TITLE
Epic 5/outdoor poi and building info

### PR DESCRIPTION
**Pull request by ThierryRomain**
- Added departments and services for every building's popup to satisfy #118 , #4 
- Added display nearest outdoor POI by types in pull-up bar #15

### CODE REVIEW

File: SD-K9/src/app/components/building-info/building-info.component.ts [hunk 1]
```diff
@@ -69,7 +71,10 @@ export class BuildingInfoComponent implements OnInit, OnDestroy {
 
   private _displayCard() {
     if (this.currentCampus !== '' && this.currentBuilding !== '') {
-      this.currentBuildingInfo = this._campusConfig[this.currentCampus]["buildings"][this.currentBuilding]['info'];
+      this.currentBuildingDepartmentsInfo = this._campusConfig[this.currentCampus]["buildings"][this.currentBuilding]['departments'];
+      this.currentBuildingServicesInfo = this._campusConfig[this.currentCampus]["buildings"][this.currentBuilding]['services'];
+      if(this.currentBuildingDepartmentsInfo == undefined) this.currentBuildingDepartmentsInfo = "no information to show";
+      if(this.currentBuildingServicesInfo == undefined) this.currentBuildingServicesInfo = "no information to show";
```
Comment (Jack-Leung) [inline SD-K9/src/app/components/building-info/building-info.component.ts:77]: Why is this necessary if you are defining the variables above it?

Comment (a-paggabao) [inline SD-K9/src/app/components/building-info/building-info.component.ts:77]: I know it looks chunky, but since we have been following this syntax of if statement, I think its best to stay consistent.
```suggestion
      if (this.currentBuildingDepartmentsInfo == undefined) {
        this.currentBuildingDepartmentsInfo = "no information to show";
      }
      if (this.currentBuildingServicesInfo == undefined) {
        this.currentBuildingServicesInfo = "no information to show";
      }
```

File: SD-K9/src/app/components/building-info/building-info.component.ts [hunk 2]
```diff
@@ -100,6 +105,17 @@ export class BuildingInfoComponent implements OnInit, OnDestroy {
     return true;
   }
 
+  toggleBuildingInfoCardTabs(e){
```
Comment (a-paggabao) [inline SD-K9/src/app/components/building-info/building-info.component.ts:108]: Is it possible to add the type of variable 'e'?
I think adding types help the code be more self-explanatory.

File: SD-K9/src/app/components/building-info/building-info.component.ts (hunk 1 above)
Comment (ThierryRomain) [reply SD-K9/src/app/components/building-info/building-info.component.ts:77]: Because this information is fetched from a JSON file and if the attributes dont exist in the object (were not specified in the file), they will be undefined.

File: SD-K9/src/app/components/building-info/building-info.component.ts [hunk 3]
```diff
@@ -121,8 +121,8 @@ export class BuildingInfoComponent implements OnInit, OnDestroy {
     return true;
   }
 
-  toggleBuildingInfoCardTabs(e){
-    let toShow = e.detail.value;
+  toggleBuildingInfoCardTabs(event : Object){
```
Comment (a-paggabao) [inline SD-K9/src/app/components/building-info/building-info.component.ts:124]: I think it's a MouseEvent, but you would need to try it out. If it still doesn't work I guess no choice, gotta use any. But at least developers will now know that this event is expected to be anything, so use at your own risk.

my source: [here](https://stackoverflow.com/questions/35966965/what-typescript-type-is-angular2-event)
